src/db/repository/postgres/OrdersRepository.ts

```diff
@@ -0,0 +1,47 @@
+import { Kysely } from 'kysely';
+
+import { DB, Orders, Orderstatus } from '../../types/generated/database';
+
+export type OrdersCreateParams = Omit<
+	Orders,
+	'id' | 'status' | 'delivery_date' | 'created_at' | 'updated_at'
+>;
+
+export default class OrdersRepository {
+	constructor(private readonly database: Kysely<DB>) {}
+
+	create(createParams: OrdersCreateParams) {
+		return this.database.insertInto('orders').values(createParams).returningAll()
+			.executeTakeFirstOrThrow;
+	}
+
+	getByDonorId(id: string) {
+		return this.database.selectFrom('orders').where('donor_id', '=', id).execute();
+	}
+
+	getByAgencyId(id: string) {
+		return this.database
+			.selectFrom('orders')
+			.innerJoin('wishcards', 'orders.wishcard_id', 'wishcards.id')
+			.innerJoin('agencies', 'wishcards.created_by', 'agencies.id')
+			.where('agencies.id', '=', id)
+			.selectAll('wishcards')
+			.execute();
+	}
```

Comment:
```suggestion
	getByAgencyId(id: string) {
		return this.database
			.selectFrom('orders')
			.innerJoin('wishcards', 'wishcards.id', 'orders.wishcard_id')
			.innerJoin('children', 'children.id', 'wishcards.child_id')
			.innerJoin('agencies', 'agencies.id', 'children.agency_id')
			.where('agencies.id', '=', id)
			.selectAll('orders')
			.execute();
	}
```

The existing query was returning wishcards and didn't have the complete joins.